### ISSUE TITLE
Local Variable Scope

### ISSUE THREAD

**Issue by HzanRsxa2959** [sannybuilder/dev#145]
_Sorry if this has been talked about before, but I didn't find any mention of it._

With the introduction of named local variables (#32), a way should be added add scope to them to avoid any potential name conflicts with opcode 0AB1.

**Comment by x87** [comment]
@HzanRsxa2959 I think I understand what you are asking for, but can you please provide a few code examples where scoped variables are used?

**Comment by HzanRsxa2959** [comment]
@x87 Here is some example code with the issues commented:
https://pastebin.com/9CyWwS38
Tested on Sanny Builder 3.7.0 released on 2021-1-30.

**Comment by x87** [comment]
another example
```
{$CLEO}

int x = 0
int y = 1
0AB1: cleo_call @fn 0

// cleo_call creates a local scope
:fn
int y = 1   // this is not the same 'y' as defined in the outer scope
cleo_return 0
```

**Comment by x87** [comment]
This is solved in 4.0 with the introduction of function syntax #263

**Comment by x87** [comment]
released in 4.0.0
